src/System.Runtime.Extensions/src/System/Diagnostics/Stopwatch.cs

```diff
@@ -29,22 +29,13 @@ public partial class Stopwatch
         // to ticks. 
         private static readonly double s_tickFrequency;
 
-        static Stopwatch()
+        static unsafe Stopwatch()
         {
-            bool succeeded = QueryPerformanceFrequency(out Frequency);
+            Frequency = QueryPerformanceFrequency();
+            IsHighResolution = true;
 
-            if (!succeeded)
-            {
-                IsHighResolution = false;
-                Frequency = TicksPerSecond;
-                s_tickFrequency = 1;
-            }
-            else
-            {
-                IsHighResolution = true;
-                s_tickFrequency = TicksPerSecond;
-                s_tickFrequency /= Frequency;
-            }
+            s_tickFrequency = TicksPerSecond;
+            s_tickFrequency /= Frequency;
```

Comment:
`s_tickFrequency = TicksPerSecond / Frequency;` ?

Reply:
This will need to be `(double)TicksPerSecond / Frequency`. It isn't clear from the code, but the logic is relying on the implicit conversion on the first step.

I will update since it makes it clear what is happening.

Reply:
Fixed.